services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/MaterializedViewManagerImpl.java

```diff
@@ -72,7 +72,22 @@ public void validate(MaterializedView materializedView) {
 	@Override
 	public void validateDefiningSql(String definingSql) {
 		ValidateArgument.requiredNotBlank(definingSql, "The definingSQL of the materialized view");
-		getQuerySpecification(definingSql);
+		
+		QueryExpression query = getQuerySpecification(definingSql);
+		
+		List<IndexDescription> indexDescriptions = getSourceTableIds(query).stream()
+				.map(sourceTableId -> tableManagerSupport.getIndexDescription(sourceTableId))
+				.collect(Collectors.toList());
+
+		IndexDescription indexDescription = 
```

Comment:
// we do not know what the ID of the MV will be yet but we need an ID for the next step.